docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,369 @@
+
+#### ShootedSeedSet Resource
+
+The `ShootedSeedSet` resource has a `selector` field that specifies how to identify ShootedSeeds it can acquire, a number of `replicas` indicating how many ShootedSeeds it should be maintaining, and a shooted seed `template` specifying the data of new ShootedSeeds it should create to meet the number of replicas criteria. A ShootedSeedSet then fulfills its purpose by creating and deleting ShootedSeeds as needed to reach the desired number. 
+
+A ShootedSeedSet is linked to its ShootedSeeds via the ShootedSeeds' `metadata.ownerReferences` field, which specifies what resource the current object is owned by. All ShootedSeeds acquired by a ShootedSeedSet have their owning ShootedSeedSet's identifying information within their `ownerReferences` field. 
+
+Example `ShootedSeedSet` resource:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: ShootedSeedSet
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      foo: bar
+  template:
+    metadata:
+      labels:
+        foo: bar
+    spec: # ShootedSeed resource
+      shoot: 
+        ...
+      seed: 
+        ...
+```
+
+#### ShootedSeedSet Controller
+
+ShootedSeedSets are reconciled by a new *shooted seed set controller* in `gardener-controller-manager`. During the reconciliation this controller creates and deletes ShootedSeeds in response to changes to the `replicas` and `selector` fields.
+
+**Note:** The introduction of the `ShootedSeedSet` resource would not require any changes to `gardenlet` or to existing `gardener-controller-manager` controllers. 
+
+#### ShootedSeedDeployments
+
+A ShootedSeedSet, similarly to a `ReplicaSet`, does not manage updates to its replicas in any way. In the future, we might introduce ShootedSeedDeployments, a higher-level concept that manages ShootedSeedSets and provides declarative updates to ShootedSeeds along with other useful features, similarly to a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). However, there is an important difference between seeds and pods or nodes in that seeds are more "heavyweight" and therefore updating a set of seeds by introducing new seeds and moving shoots to them tends to be much more complex, time-consuming, and prone to failures compared to updating the seeds "in place". Furthermore, updating seeds in this way depends on a mature implementation of [GEP-7: Shoot Control Plane Migration](07-shoot-control-plane-migration.md), which is not available right now. Therefore, ShootedSeedDeployments are not described in detail here, as properly dealing with the challenges mentioned requires a separate GEP.
+
+#### Deleting ShootedSeeds
```

Comment:
I think this section belongs exactly where it is, since what it describes are mainly challenges with scaling-in ShootedSeedSets, not deleting ShootedSeeds per se. A ShootedSeed that has Shoots scheduled onto it can't be deleted, the deletion simply fails as it does today, and that's it. This perhaps deserves a sentence in the `ShootedSeed` section, but not an entire section. The title however could be improved, instead of *Deleting ShootedSeeds* it should be *Scaling-in ShootedSeedSets*.

Reply:
A sentence has been added to the **ManagedSeed Controller** section:

> A `ManagedSeed` that is in use by shoots cannot be deleted, unless the shoots are either deleted or moved to other seeds first. The managed seed controller ensures that this is the case.

The above section has been renamed to **Scaling-in ManagedSeedSets** but otherwise kept mostly as it is except adding the suggestions from @rfranzke above.